packages/core/test/unit/sql/literal.test.ts

```diff
@@ -0,0 +1,152 @@
+        language: 'icelandic',
+        pg_rating: { dk: 'G' },
+      },
+      another_json_field: { x: 1 },
+    };
+
+    expectsql(() => queryGenerator.escape(json(conditions)), {
+      postgres: `("metadata"->'language' = '"icelandic"' AND "metadata"#>ARRAY['pg_rating','dk'] = '"G"') AND "another_json_field"->'x' = '1'`,
+      sqlite: `(json_extract(\`metadata\`,'$.language') = '"icelandic"' AND json_extract(\`metadata\`,'$.pg_rating.dk') = '"G"') AND json_extract(\`another_json_field\`,'$.x') = '1'`,
+      mariadb: `(json_compact(json_extract(\`metadata\`,'$.language')) = '"icelandic"' AND json_compact(json_extract(\`metadata\`,'$.pg_rating.dk')) = '"G"') AND json_compact(json_extract(\`another_json_field\`,'$.x')) = '1'`,
+      mysql: `(json_extract(\`metadata\`,'$.language') = CAST('"icelandic"' AS JSON) AND json_extract(\`metadata\`,'$.pg_rating.dk') = CAST('"G"' AS JSON)) AND json_extract(\`another_json_field\`,'$.x') = CAST('1' AS JSON)`,
+    });
+  });
+
+  it('supports the json path notation', () => {
+    const path = 'metadata.pg_rating.dk';
+
+    expectsql(() => queryGenerator.escape(json(path)), {
+      postgres: `"metadata"#>ARRAY['pg_rating','dk']`,
+      mariadb: `json_compact(json_extract(\`metadata\`,'$.pg_rating.dk'))`,
+      'sqlite mysql': `json_extract(\`metadata\`,'$.pg_rating.dk')`,
+    });
+  });
+
+  it('supports numbers in the dot notation', () => {
+    expectsql(queryGenerator.escape(json('profile.id.0.1')), {
+      postgres: `"profile"#>ARRAY['id','0','1']`,
+      mariadb: `json_compact(json_extract(\`profile\`,'$.id."0"."1"'))`,
+      'sqlite mysql': `json_extract(\`profile\`,'$.id."0"."1"')`,
+    });
+  });
+
+  it('can take a value to compare against', () => {
+    const path = 'metadata.pg_rating.is';
+    const value = 'U';
+
+    expectsql(() => queryGenerator.escape(json(path, value)), {
+      postgres: `"metadata"#>ARRAY['pg_rating','is'] = '"U"'`,
+      sqlite: `json_extract(\`metadata\`,'$.pg_rating.is') = '"U"'`,
+      mariadb: `json_compact(json_extract(\`metadata\`,'$.pg_rating.is')) = '"U"'`,
+      mysql: `json_extract(\`metadata\`,'$.pg_rating.is') = CAST('"U"' AS JSON)`,
+    });
+  });
+
+  // TODO: add a way to let `where` know what the type of the value is in raw queries
+  // it('accepts a condition object', () => {
+  //   expectsql(queryGenerator.escape(json({ id: 1 })), {
+  //     postgres: `"id" = '1'`,
+  //   });
+  // });
+  //
+  // it('column named "json"', () => {
+  //   expectsql(queryGenerator.escape(where(json('json'), Op.eq, {})), {
+  //     postgres: `("json"#>>'{}') = '{}'`,
+  //   });
+  // });
+
+  it('accepts a nested condition object', () => {
+    expectsql(queryGenerator.escape(json({ profile: { id: 1 } })), {
+      postgres: `"profile"->'id' = '1'`,
+      sqlite: `json_extract(\`profile\`,'$.id') = '1'`,
+      mariadb: `json_compact(json_extract(\`profile\`,'$.id')) = '1'`,
+      mysql: `json_extract(\`profile\`,'$.id') = CAST('1' AS JSON)`,
```

Comment:
Good question :) It's something I discovered while running integration tests
MySQL can compare JSON to JSON and text to text, but it can't compare text to JSON, so we have to cast the value

Reply:
Interesting that it works on other dialects but not here